### PR TITLE
Issue 6836: Temporarily revert PR 6528

### LINKED ISSUE #6836: Temporarily Revert PR 6528
**Describe the bug**
Few regressions noticed with PR #6528 in case of multiple restarts of Segment Stores Pods. This PR can be temporarily reverted.

### PR DESCRIPTION
**Pull request by shshashwat**
Signed-off-by: Shashwat Sharma <shashwat_sharma@dell.com>

**Change log description**  
This PR reverts changes from PR #6528 due to introduction of a regression.

**Purpose of the change**  
Fixes #6836 

**What the code does**  
reverts changes from PR #6528 because of a regression found in a scenario

### CODE REVIEW

**Comment by shshashwat** [comment]
System test successful on :- TPC-1 build no -4700

File: common/src/test/java/io/pravega/common/util/RetryTests.java [hunk 1]
```diff
@@ -255,6 +254,11 @@ public void retryIndefiniteTest() throws ExecutionException, InterruptedExceptio
                     }
                 }), pool).get();
         assert i.get() == 10;
+        try {
+            Retry.indefinitelyWithExpBackoff(0, 0, 0, e -> i.getAndIncrement());
```
Comment (RaulGracia) [inline common/src/test/java/io/pravega/common/util/RetryTests.java:258]: Use `AssertThrows` instead of this try/catch + assertion construct.

**Review by RaulGracia — COMMENTED** [review]
LGTM, just one minor comment.

File: common/src/test/java/io/pravega/common/util/RetryTests.java (hunk 1 above)
Comment (shshashwat) [reply common/src/test/java/io/pravega/common/util/RetryTests.java]: Added the `AssertThrows` and removed `try/catch`

**Review by RaulGracia — APPROVED** [review]

**Pull request merged**